src/components/upload-promotion/UploadPromotionForm.tsx

```diff
@@ -113,7 +116,19 @@ export default function UploadPromotionForm(): ReactElement {
       schedules: getSchedules(),
       startDate: formValues.datesEffective[0],
       userId: UserService.userId,
-    });
+    };
+
+    PromotionService.postPromotion(promotionDTO)
+      .then(() => {
+        history.push('/promotions');
```

Comment:
Interesting! I never hear about `useHistory`, very cool 🆒

Reply:
Oh yeah it's a custom hook from the `react-router-dom` library that just lets you manipulate the URL 😛